mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -71,6 +71,26 @@ private[recommendation] trait ALSModelParams extends Params with HasPredictionCo
 
   /** @group getParam */
   def getItemCol: String = $(itemCol)
+
+  /**
+   * Param for strategy for dealing with unknown or new users/items at prediction time.
+   * This may be useful in cross-validation or production scenarios, for handling user/item ids
+   * the model has not seen in the training data.
+   * Supported values:
+   * - "nan": predicted value for unknown ids will be NaN.
+   * - "drop": rows in the input DataFrame containing unknown ids will be dropped.
```

Comment:
Fair enough. Will reword this a bit.
On Wed, 11 May 2016 at 17:25, Seth Hendrickson <notifications@github.com>
wrote:

> In mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala
> <https://github.com/apache/spark/pull/12896#discussion_r62869002>:
>
> > @@ -71,6 +71,26 @@ private[recommendation] trait ALSModelParams extends Params with HasPredictionCo
> >
> >    /** @group getParam */
> >    def getItemCol: String = $(itemCol)
> > +
> > +  /**
> > +   * Param for strategy for dealing with unknown or new users/items at prediction time.
> > +   * This may be useful in cross-validation or production scenarios, for handling user/item ids
> > +   * the model has not seen in the training data.
> > +   * Supported values:
> > +   * - "nan": predicted value for unknown ids will be NaN.
> > +   * - "drop": rows in the input DataFrame containing unknown ids will be dropped.
>
> "will be dropped." -> "will not be included in the recommendations."
>
> It could sound like we are modifying the input dataframe, but we are not.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/spark/pull/12896/files/b71dcaca39d5f10b484932eaf25e290c7796d93b#r62869002>
>